src/intrinsics/ecma262/BooleanPrototype.js

```diff
@@ -10,20 +10,29 @@
 /* @flow */
 
 import type { Realm } from "../../realm.js";
-import { ObjectValue, StringValue } from "../../values/index.js";
+import { ObjectValue, StringValue, AbstractValue, BooleanValue } from "../../values/index.js";
 import { thisBooleanValue } from "../../methods/to.js";
+import { TypesDomain, ValuesDomain } from "../../domains/index.js";
+import buildExpressionTemplate from "../../utils/builder.js";
 
 export default function(realm: Realm, obj: ObjectValue): void {
   // ECMA262 19.3.1
   obj.$BooleanData = realm.intrinsics.false;
 
   // ECMA262 19.3.3.3
   obj.defineNativeMethod("toString", 0, context => {
-    // 1. Let b be ? thisBooleanValue(this value).
-    let b = thisBooleanValue(realm, context);
+    if (context instanceof AbstractValue && context.getType() === BooleanValue) {
+      const codeTemplate = "(A).toString()";
+      return realm.createAbstract(new TypesDomain(StringValue), ValuesDomain.topVal, [context], ([a]) =>
+        buildExpressionTemplate(codeTemplate)(realm.preludeGenerator)({ A: a })
+      );
+    } else {
```

Comment:
Since the if branch always returns you don't need an else branch here. I prefer to omit such else branches so that the lines don't get too long because of excessive indentation.